### DISCUSSION TITLE
How to generate a TypeScript REST client from proto files with gRPC transcoding?

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/aspnetcore#59492] [Q&A]
While gRPC transcoding supports generating OpenApi docs, I'd like to know if there's a more direct way to do this. Note this would be intended to be used in the browser. Thanks.

**Comment by wilsonneto-swe** [comment]
I don't think this question is related to aspnetcore. Also, I would not expect the .Net framework to auto-generate TS clients.

I think that you can use Kiota over the OAS generated:
https://learn.microsoft.com/en-us/openapi/kiota/quickstarts/typescript

Or you can use gapic-gen:
https://github.com/googleapis/gapic-generator-typescript

I hope it helps!

**Reply by alrz** [reply]
>I don't think this question is related to aspnetcore. 

I kind of agree but I think this could be also a missing piece. kiota works on openapi and majority of client generators are grpc clients not rest. gRPC transcoding is an efficient, self-contained alternative mechanism to grpc-web which requires a proxy. I think a rest client generators which operates directly on http.proto would be very useful when you are using transcoding on the server.

**Reply by alrz** [reply]
Tbh I wasn't sure where else I could post this question as every tool I looked into is not exactly supposed to work with transcoding.

**Reply by wilsonneto-swe** [reply]
Good point! Got it; 
Let's see if other folks have better info or suggestions
